src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs

```diff
@@ -70,29 +72,33 @@ private async Task<ResultWrapper<PayloadStatusV1>> NewPayload(ExecutionPayload e
             return ResultWrapper<PayloadStatusV1>.Fail(error, ErrorCodes.InvalidParams);
         }
 
-        if (await _locker.WaitAsync(_timeout))
+        using (_gcKeeper.TryStartNoGCRegion())
         {
-            Stopwatch watch = Stopwatch.StartNew();
-            try
+            if (await _locker.WaitAsync(_timeout))
             {
-                return await _newPayloadV1Handler.HandleAsync(executionPayload);
+                Stopwatch watch = Stopwatch.StartNew();
+                try
+                {
+                    return await _newPayloadV1Handler.HandleAsync(executionPayload);
+                }
+                catch (Exception exception)
+                {
+                    if (_logger.IsError) _logger.Error($"engine_newPayloadV{version} failed: {exception}");
+                    return ResultWrapper<PayloadStatusV1>.Fail(exception.Message);
+                }
+                finally
+                {
+                    watch.Stop();
+                    Metrics.NewPayloadExecutionTime = watch.ElapsedMilliseconds;
+                    _gcKeeper.ScheduleGC();
```

Comment:
So... why not call this as part of the dispose of the GC region?

Reply:
leftover from first design